client/landing/jetpack-cloud/sections/backups/download/index.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import React, { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { BackupProgress } from './types';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteTitle } from 'state/sites/selectors';
+import { rewindBackup } from 'state/activity-log/actions';
+import { useLocalizedMoment } from 'components/localized-moment';
+import Confirm from './confirm';
+import getBackupProgressForRewindId from 'state/selectors/get-backup-progress-for-rewind-id';
+import getRewindBackupProgressRequestStatus from 'state/selectors/get-rewind-backup-progress-request-status';
+import InProgress from './in-progress';
+import QueryRewindBackupStatus from 'components/data/query-rewind-backup-status';
+import Ready from './ready';
+import Placeholder from './placeholder';
+import Error from './error';
+
+interface Props {
+	rewindId: string;
+}
+
+const BackupRestorePage = ( { rewindId }: Props ) => {
```

Comment:
Why use a functional component vs a one that is connected to the state?

Reply:
this is a functional component that is very much connected to state via `useSelector` and `useDispatch` below!

I prefer the logic of a functional component in all cases because the logic needs to be apparent in the flow of the function, instead of split across multiple lifecycle functions that I can never remember the order of. With hooks everything a class component can do is possible in a functional component.

Reply:
Thanks for the explanation. Do you think we should follow this pattern going forward?

Reply:
yup, that is the path React is taking, so I think it is the best interest of our project to follow!